RawSpeed/DngDecoder.cpp

```diff
@@ -230,17 +238,23 @@ RawImage DngDecoder::decodeRawInternal() {
       } catch (TiffParserException) {
         ThrowRDE("DNG Decoder: Unsupported format, uncompressed with no strips.");
       }
-    } else if (compression == 7 || compression == 0x884c) {
+    } else if (compression == 7 || compression == 8 || compression == 0x884c) {
       try {
         // Let's try loading it as tiles instead
 
-        mRaw->setCpp(raw->getEntry(SAMPLESPERPIXEL)->getInt());
         mRaw->createData();
 
-        if (sample_format != 1)
-           ThrowRDE("DNG Decoder: Only 16 bit unsigned data supported for compressed data.");
+        if (compression == 8 && sample_format != 3)
+           ThrowRDE("DNG Decoder: Only float format is supported for deflate-compressed data.");
+        else if (compression != 8 && sample_format != 1)
```

Comment:
But `compression != 8` only says that it is not deflate compression, not that it is jpeg compression?